dangerzone/isolation_provider/base.py

```diff
@@ -88,8 +88,11 @@ def convert(
             log.exception(
                 f"An exception occurred while converting document '{document.id}'"
             )
-            self.print_progress(document, True, str(e), 0)
-            document.mark_as_failed()
+            try:
+                self.print_progress(document, True, str(e), 0)
+                document.mark_as_failed()
+            except Exception as e:
+                log.exception(f"Error occured while marking document as failed '{document.id}'")
```

Comment:
Adding a try...except block here is not necessary, since the plan is to catch exceptions even higher up. I would ditch this change.

Reply:
Removed this change